### PR TITLE
Add endpoint fallback support

### PR DESCRIPTION
**Pull request by ilfa**

### CODE REVIEW

**Review by necipallef — APPROVED** [review]

**Comment by necipallef** [comment]
Do we expose the [default endpoint](https://dev.fingerprint.com/docs/js-agent#endpoint) in the SDK? If not, I'd like to see it exposed in this PR as well. @ilfa

**Comment by ilfa** [comment]
> Do we expose the [default endpoint](https://dev.fingerprint.com/docs/js-agent#endpoint) in the SDK? If not, I'd like to see it exposed in this PR as well. @ilfa

As I understand, we don't.
@Alexey-Verkhovsky you should know better.

File: ios/RNFingerprintjsPro.swift [hunk 1]
```diff
@@ -18,9 +18,9 @@ class RNFingerprintjsPro: NSObject {
     @objc
     static func requiresMainQueueSetup() -> Bool { false }
 
-    @objc(init:region:endpoint:extendedResponseFormat:pluginVersion:)
-    public required init(_ apiToken: String, _ region: String? = "us", _ endpoint: String? = nil, _ extendedResponseFormat: Bool = false, _ pluginVersion: String) {
-        let region = RNFingerprintjsPro.parseRegion(region, endpoint: endpoint)
+    @objc(init:region:endpoint:endpointFallbacks:extendedResponseFormat:pluginVersion:)
+    public required init(_ apiToken: String, _ region: String? = "us", _ endpoint: String?, _ fallbackEndpointUrls: [String], _ extendedResponseFormat: Bool = false, _ pluginVersion: String) {
```
Comment (mgutski) [inline ios/RNFingerprintjsPro.swift:22]: Here, for backward compatibility, I think the `fallbackEndpointUrls` parameter should have a default value of `[]`.

Comment (mgutski) [reply ios/RNFingerprintjsPro.swift:22]: Also, for naming consistency with the Flutter library, I think the parameter should be renamed to `endpointFallbacks`.

Comment (mgutski) [reply ios/RNFingerprintjsPro.swift:22]:
```suggestion
    public required init(_ apiToken: String, _ region: String? = "us", _ endpoint: String?, _ endpointFallbacks: [String] = [], _ extendedResponseFormat: Bool = false, _ pluginVersion: String) {
```

File: ios/RNFingerprintjsPro.swift [hunk 2]
```diff
@@ -62,7 +62,7 @@ class RNFingerprintjsPro: NSObject {
         }
 
 
-    private static func parseRegion(_ passedRegion: String?, endpoint: String?) -> Region {
+    private static func parseRegion(_ passedRegion: String?, endpoint: String?, fallbackEndpointUrls: [String]?) -> Region {
```
Comment (mgutski) [inline ios/RNFingerprintjsPro.swift:65]: For naming consistency:
```suggestion
    private static func parseRegion(_ passedRegion: String?, endpoint: String?, endpointFallbacks: [String]) -> Region {
```
Also, the added parameter does not need to be an optional.

File: ios/RNFingerprintjsPro.swift (hunk 1 above)
Comment (mgutski) [reply ios/RNFingerprintjsPro.swift:22]: Ah, now I see that the default value is provided on the TypeScript level in the `src/FingerprintJsProAgent.ts` file, so it's just the naming part that needs to be addressed there.

**Comment by ilfa** [comment]
@mgutski thank you for the review!
Fixed.

**Review by mgutski — APPROVED** [review]

**Review by Alexey-Verkhovsky — APPROVED** [review]

**Review by necipallef — APPROVED** [review]

**Pull request merged**